pkg/aws/ec2/api/helper.go

```diff
@@ -594,3 +619,81 @@ func (h *ec2APIHelper) DetachAndDeleteNetworkInterface(attachmentID *string, nwI
 	}
 	return nil
 }
+
+func (h *ec2APIHelper) GetSecurityGroupIdsForSecurityGroupNames(securityGroupNames []string) ([]string, error) {
+	sgIds := make([]string, 0)
+	if len(securityGroupNames) < 1 {
+		return sgIds, nil
+	}
+	sgNamesNotInCache := make([]string, 0)
+	for _, sgName := range securityGroupNames {
+		if sgId, ok := h.securityGroupNameToIdMap[sgName]; !ok {
+			sgNamesNotInCache = append(sgNamesNotInCache, sgName)
+		} else {
+			sgIds = append(sgIds, sgId)
+		}
+	}
+	if len(sgIds) == len(securityGroupNames) {
+		//All SGNames found in cache
+		return sgIds, nil
+	}
+	filters := []*ec2.Filter{
+		{
+			Name:   aws.String("group-name"),
+			Values: aws.StringSlice(sgNamesNotInCache),
+		},
+		{
+			Name:   aws.String("vpc-id"),
+			Values: aws.StringSlice([]string{h.workerNodeVpcId}),
+		},
+	}
+	input := &ec2.DescribeSecurityGroupsInput{Filters: filters}
+	foundSgNames := map[string]bool{}
+	for {
+		output, err := h.ec2Wrapper.DescribeSecurityGroups(input)
```

Comment:
Can we consider using the `GetSecurityGroupsForVpc` API call instead of `DescribeSecurityGroups` here?

Reply:
Yup let me look into this